research-hub-web/src/app/app.module.ts

```diff
@@ -41,20 +65,59 @@ import { HomeModule } from './components/home/home.module';
     RoutingModule,
     ErrorPagesModule,
     HttpClientModule,
-    // MaterialModule,
     BrowserAnimationsModule,
     HttpClientModule,
     FlexLayoutModule,
-    HomeModule
+    HomeModule,
+    HomeModule,
+    ApolloModule,
+    HttpLinkModule,
+    ErrorPagesModule
   ],
   entryComponents: [],
   providers: [
     HeaderService,
     SearchBarService,
     AppComponentService,
     SearchFiltersService,
-    { provide: CognitoConfigService, useClass: AppAuthConfigService }],
+    { provide: CognitoConfigService, useClass: AppAuthConfigService },
+    { provide: StorageService, useClass: AppStorageService },
+  ],
   bootstrap: [AppComponent]
 })
 export class AppModule {
+  constructor(private apollo: Apollo, private httpLink: HttpLink, private loginService: LoginService, private router: Router) {
+
+    /**
+     * Creates global Apollo client and error handler.
+     */
+    const errorLink = onError(({ graphQLErrors, networkError }) => {
+      if (graphQLErrors) {
+        graphQLErrors.map(({ message, locations, path }) =>
+          console.log(`[GraphQL error]: Message: ${message}, Location: ${location}, Path: ${path}`),
+        );
+      }
+      if (networkError) {
+        console.error('Network error detected');
+        console.log(`[Network error]: ${JSON.stringify(networkError, null, 2)}`);
+
+        // Check for SSO protected error
+        if (networkError['error']['errors'][0]['extensions']['code'] === 'UNAUTHENTICATED') {
+          console.log('SSO Authentication Error Detected at route:', this.router.url);
+          this.loginService.doLogin(this.router.url);
+        }
```

Comment:
Console log that can be removed

Reply:
Well spotted! Removed that whole `if (graphqlError) {` section as we're not doing anything about that Removed the remaining `console.log`s 🤺